apps/src/javalab/JavabuilderConnection.js

```diff
@@ -104,6 +104,19 @@ export default class JavabuilderConnection {
 
   // Send a message across the websocket connection to Javabuilder
   sendMessage(message) {
-    this.socket.send(message);
+    if (!this.socket) {
+      console.error(`[error] The connection has closed.`);
+    } else {
+      this.socket.send(message);
+    }
+  }
+
+  // Closes web socket connection
+  closeConnection() {
+    if (!this.socket) {
+      console.error(`[error] There is no web socket connection.`);
```

Comment:
similar question here--can a user hit this in a normal execution scenario?

Reply:
I think I have a similar answer- there will still be an error if we assume there is a socket. It's possible to create a JavabuilderConnection without a socket, so it seems we shouldn't try to send signals along a socket without checking for its existence first. It's also possible I'm unclear on how sockets work!